src/lib/Liblog/pbs_log.c

```diff
@@ -275,15 +271,15 @@ log_get_tls_data(void)
 int
 log_mutex_lock()
 {
-	long log_lock;
-	if ((log_lock = (long)pthread_getspecific(pbs_log_tls_key)) != 0)
+	void *log_lock;
+	if ((log_lock = pthread_getspecific(pbs_log_tls_key)) != 0)
 		return -1;
 
 	if (pthread_mutex_lock(&log_mutex) != 0)
 		return -1;
 	
 	log_lock = 1;
```

Comment:
I think you can use log_lock = &log_lock here. That should provide a unique non-NULL value.